pir/cpp/server_test.cpp

```diff
@@ -115,6 +137,67 @@ TEST_F(PIRServerTest, TestCorrectness) {
   ASSERT_EQ(result[0], db_[desired_index]);
 }
 
+TEST_F(PIRServerTest, TestCorrectnessD2) {
+  SetUpDB(87, 2);
+  auto client = PIRClient::Create(pir_params_).ValueOrDie();
+  const size_t desired_index = 42;
+  auto request = client->CreateRequest({desired_index}).ValueOrDie();
+  auto response = server_->ProcessRequest(request).ValueOrDie();
+  auto result = client->ProcessResponse(response).ValueOrDie();
+
+  ASSERT_EQ(result.size(), 1);
+  ASSERT_EQ(result[0], db_[desired_index]);
+}
+
+TEST_F(PIRServerTest, TestCorrectnessLargeValues) {
+  SetUpStringDB(10, 9728);
+
+  auto client = PIRClient::Create(pir_params_).ValueOrDie();
+  const size_t desired_index = 5;
+
+  auto request = client->CreateRequest({desired_index}).ValueOrDie();
+  auto response = server_->ProcessRequest(request).ValueOrDie();
+  auto result = client->ProcessResponseString(response).ValueOrDie();
+
+  ASSERT_THAT(result, SizeIs(1));
+  ASSERT_GE(result[0].size(), string_db_[desired_index].size());
+  EXPECT_EQ(result[0].substr(0, string_db_[desired_index].size()),
+            string_db_[desired_index]);
+  EXPECT_THAT(result[0].substr(string_db_[desired_index].size()),
+              testing::Each(0));
+}
+
+void print_hex_string(const string& desc, const string& s) {
+  cout << desc;
+  for (auto c : s) {
+    cout << std::hex << std::setw(2) << std::setfill('0')
+         << (uint16_t)(uint8_t)c;
+  }
+  cout << endl;
+}
+
+TEST_F(PIRServerTest, DISABLED_TestCorrectnessLargeValuesD2) {
+  SetUpStringDB(12, 32, 2);
+
+  auto client = PIRClient::Create(pir_params_).ValueOrDie();
+  const size_t desired_index = 7;
+
+  auto request = client->CreateRequest({desired_index}).ValueOrDie();
+  auto response = server_->ProcessRequest(request).ValueOrDie();
+  auto result = client->ProcessResponseString(response).ValueOrDie();
+
+  ASSERT_THAT(result, SizeIs(1));
+  ASSERT_GE(result[0].size(), string_db_[desired_index].size());
+  print_hex_string("Expected: ", string_db_[desired_index]);
+  print_hex_string("Actual  : ",
+                   result[0].substr(0, string_db_[desired_index].size()));
+
+  EXPECT_EQ(result[0].substr(0, string_db_[desired_index].size()),
+            string_db_[desired_index]);
+  // EXPECT_THAT(result[0].substr(string_db_[desired_index].size()),
```

Comment:
why is this commented?

Reply:
Fixed. Test is disabled, so this isn't used at the moment. I'd commented because when enabled this adds a lot of noise to the test.